src/backends/cpu.jl

```diff
@@ -41,7 +41,15 @@ end
 function __run(obj, ndrange, iterspace, args, dependencies)
     return Threads.@spawn begin
         if dependencies !== nothing
-            Base.sync_end(map(e->e.task, dependencies))
+            cpu_tasks = Core.Task[]
+            for event in dependencies
+                if event isa CPUEvent
+                    push!(cpu_tasks, event.task)
+                else
+                    wait(event)
```

Comment:
```suggestion
                    wait(event, ()->yield())
```

I am worried about busy-waiting on the GPU causing us the block a thread from making progress. So we should pass a CPU yield as remedy.

Reply:
Done, and added a yield in the cuda backend as well.